### ISSUE TITLE
Having trouble compiling (jump-patch.h)

### ISSUE THREAD

**Issue by soymjolk** [comex/substitute#24]
Hey there. I'm playing around with this and I'm unable to get it to finish. The error it leaves me with is:

In file included from ./lib/hook-functions.c:7:
./lib/arm/jump-patch.h:17:40: error: incompatible integer to pointer conversion
      initializing 'void *' with an expression of type 'uint_tptr'
      (aka 'unsigned int') [-Werror,-Wint-conversion]
    struct assemble_ctx actx = {codep, pc, arch.pc_low_bit, 0xe};
                                                              ^~
1 error generated.

Thanks for the cool stuff!

**Comment by comex** [comment]
yeah, that looks broken, sorry.  I haven't touched this codebase in quite a while.

I think it might work to change it to
```c
struct assemble_ctx actx = {codep, *codep, pc, arch.pc_low_bit, 0xe};
```

**Comment by soymjolk** [comment]
Hey, that seems to have worked! Ended up getting this after throwing that
line in:

Undefined symbols for architecture arm64:
  "__dyld_get_all_image_infos", referenced from:
      _inspect_dyld in find-syms.o
      _substitute_dlopen_in_pid in inject.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see
invocation)
make: *** [out/libsubstitute.dylib] Error 1

I'm gonna play around some more and see if I can fix it, will update if
anything happens. Thanks again!

**Comment by shinvou** [comment]
@soymjolk That happens because, well 

> ld: symbol(s) not found for architecture arm64

these symbols are gone in iOS 11. See #22 for a fix (although I'd use ifdef for the fix, seems cleaner).

Maybe @comex can just update these two things and commit them. I'd probably take like five minutes.
